editors/code/src/extension.ts

```diff
@@ -133,3 +138,11 @@ export function deactivate(): Thenable<void> {
     }
     return Server.client.stop();
 }
+
+async function reloadServer(startServer: () => void) {
+    if (Server.client != null) {
+        vscode.window.showInformationMessage('Reloading rust-analyzer...');
+        await Server.client.stop();
```

Comment:
So this sends notification/$exit, but doesn't close stdin?

Reply:
Nope, the vscode base language client does not attempt to close stdin as far as I can see, though it does try to kill the process (and fails, as mentioned above).

Reply:
Filed https://github.com/Microsoft/vscode-languageserver-node/issues/485